sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/LocationCache.java

```diff
@@ -272,6 +272,23 @@ public boolean shouldRefreshEndpoints(Utils.ValueHolder<Boolean> canRefreshInBac
             return false;
         }
     }
+
+    public String getRegionName(URI locationEndpoint, com.azure.cosmos.implementation.OperationType operationType) {
+        Utils.ValueHolder<String> regionName = new Utils.ValueHolder<>();
+        if (operationType.isWriteOperation()) {
+            if (Utils.tryGetValue(this.locationInfo.regionNameByWriteEndpoint, locationEndpoint, regionName)) {
+                return regionName.v;
+            }
+        } else {
+            if (Utils.tryGetValue(this.locationInfo.regionNameByReadEndpoint, locationEndpoint, regionName)) {
+                return regionName.v;
+            }
+        }
+
+        //If preferred list is not set, locationEndpoint will be default endpoint, so return the hub region
+        return this.locationInfo.availableWriteEndpointByLocation.keySet().iterator().next();
+    }
```

Comment:
does the new method `public String getRegionName(URI locationEndpoint, operationType)` guarantee that always give accurate region name in any scenario?

I would like you to think about any edge cases :-)